Add sum stat to basicstats aggregator

cpacey opened the pull request:
Implements #3467.  `sum` is not included by default to maintain backwards compatibility.

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

cpacey commented on plugins/aggregators/basicstats/basicstats.go:
```diff
@@ -146,6 +147,9 @@ func (m *BasicStats) Push(acc telegraf.Accumulator) {
 				fields[k+"_mean"] = v.mean
 			}
+			if config.sum {
+				fields[k+"_sum"] = (v.mean * v.count)
```
This seemed the simplest way to implement it.

JeffAshton replied:
And cheapest for backwards compatibility case

cpacey commented on plugins/aggregators/basicstats/basicstats_test.go:
```diff
@@ -357,3 +434,26 @@ func TestBasicStatsWithUnknownStat(t *testing.T) {
+// stdev, and s2.  We purposely exclude sum for backwards compatability,
+// otherwise user's working systems will suddenly (and surprisingly) start
+// capturing sum without their input.
+func TestBasicStatsWithDefaultStats(t *testing.T) {
```
This might be overkill, but I like the idea of verifying "default configuration" separately from verifying "aggregations are working".

JeffAshton commented:
Hey @danielnelson , I would love to use this feature to help simplify my continuous queries.  Do you have time to take a look?

danielnelson commented:
One thing I'm worried about is that we might see some weird floating point artifacts that wouldn't appear if we kept a running sum:
```
cpu value=1 1519251291871635057
cpu value=1 1519251291871803032
cpu value=1 1519251291871857634
cpu value=1 1519251291871884218
cpu value=2 1519251291871907024
cpu value=1 1519251291871931726
cpu value_sum=6.999999999999999 1519251294000000000
```

cpacey commented:
@danielnelson I'll switch to tracking `sum` as we go.  Would you like to keep tracking `mean` as it's done now, or would you want to switch to calculating the mean based on `sum` and `count`?  The former simplifies the code change and reduces the likelihood of changing previous behaviour; the latter simplifies the code and reduces calculation cost.

danielnelson commented:
Let's switch to calculating the mean based on sum and count.

cpacey commented:
@danielnelson I've switched to keeping a running sum (and added a test for the floating-point problem you were concerned about).  However, when I tried to remove tracking the mean, the naive approach added another floating-point divide when calculating variance (example [here](https://github.com/cpacey/telegraf/pull/1)).  Do we care about a potential performance impact of this?  I haven't measured anything yet - I wanted to get a sense of whether you'd be concerned before investing time.

If you are concerned, options I see are to continue tracking the mean, or rearranging the variance equations to avoid the extra floating-point divide, but at the cost of an extra floating-point multiplication (example [here](https://github.com/cpacey/telegraf/pull/2)).

danielnelson commented:
In the first example, doesn't it equal out since we can remove this divide?
```
mean = mean + delta/n
```

cpacey commented:
Not without changing the `stdev` calculation, which I'm not excited to do.  The [online variance algorithm](https://en.m.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online_algorithm) produces its estimate by using both the _current_ mean and the _previous_ mean.  In the current code, the previous mean is tracked, and the new mean is calculated with a single divide (`mean + delta/n`).  If we stop tracking the mean, then we need to compute _both_ the previous mean and the current mean, which (naively) requires 2 divides: either `oldSum / oldCount` and `newSum / newCount`; or `oldSum / oldCount` and `oldMean + delta/n`.

(Note that I'd be very happy to be wrong about this.)

cpacey commented:
@danielnelson, could we keep tracking `mean` and merge as-is?
